### DISCUSSION TITLE
Having problems with server & client component design pattern

### DISCUSSION THREAD

**Discussion by Scala-Tech** [vercel/next.js#60126] [Help]
### Summary

Suppose i am working on Navbar component which have three columns
1. Navbar Brand log  **(should render in server ) ---> Server Component**
2. Navbar links  **(should render in server ) ---> Server Component**
3 Profiles Pictures and dropdown.  **(should render in client ) ---> use redux loggedIn user details state**

i think  i will get issues in:

inorder to use redux state in navbar, i have to wrap the 'Navbar' component by react-redux 'Provider' which makes my 'Navbar' component to 'client component' right ? then i will not be able to use server component inside 'Navbar ' right ?

Can anyone please help me to solve this problems ?

![image](https://github.com/vercel/next.js/assets/154963836/1d8bf7d8-de56-4c58-8558-4d20f94f437e)

**Comment by heysahilsingh** [comment]
Using server component as a child of a client component, doesn’t make server component a client component. Means if you are using a server comp. as a children of client comp., the server comp. will remain the server comp.

**Reply by dw820** [reply]
Thanks for explaining, I have a follow-up question on this, 

According to NextJs [doc](https://nextjs.org/docs/app/building-your-application/rendering/client-components#using-client-components-in-nextjs), it mentioned:

> This means that by defining a "use client" in a file, all other modules imported into it, including child components, are considered part of the client bundle.

In this case, the server component passed as a children of the client component, It remains a server component but is considered as the client's bundle? Is that correct? And what's the implication of being as the client's bundle?

**Reply by heysahilsingh** [reply]
Please go through this [doc](https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns), this might help in understanding the concept of the server component being used as a child of the client component.
